aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -258,47 +257,47 @@ namespace {
     }
   }
 
-  template <typename scalar_t>
+  template <typename scalar_t, typename index_t>
   C10_LAUNCH_BOUNDS_1(1024)
   __global__ void grid_sampler_2d_backward_kernel(
-      const int nthreads,
-      TensorInfo<scalar_t, int> grad_output,
-      TensorInfo<scalar_t, int> input,
-      TensorInfo<scalar_t, int> grid,
-      TensorInfo<scalar_t, int> grad_input,  // initialized to zeros
-      TensorInfo<scalar_t, int> grad_grid,   // initialized to empty
+      const index_t nthreads,
+      TensorInfo<scalar_t, index_t> grad_output,
+      TensorInfo<scalar_t, index_t> input,
+      TensorInfo<scalar_t, index_t> grid,
+      TensorInfo<scalar_t, index_t> grad_input,  // initialized to zeros
+      TensorInfo<scalar_t, index_t> grad_grid,   // initialized to empty
```

Comment:
We should definitely add tests for the backward since we are adding a path to do backward with 64-bit indexing